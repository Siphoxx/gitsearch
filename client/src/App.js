// client/src/App.js
import React from "react";
import { Route, Routes } from "react-router-dom";
import UserSearch from "./components/UserSearch";
import UserDetails from "./components/UserDetails";

const App = () => {
  return (
    <div>
      <h1>GitHub UserSearch</h1>
      <Routes>
        <Route path="/" element={<UserSearch />} />
        <Route path="/user/:username" element={<UserDetails />} />
      </Routes>
    </div>
  );
};

export default App;
