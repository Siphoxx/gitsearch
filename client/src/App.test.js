// src/App.test.js
import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import App from "./App";

test("renders GitHub UserSearch heading", () => {
  const { getByText, asFragment } = render(
    <Router>
      <App />
    </Router>
  );

  // Functional test: Check if the heading is in the document
  const headingElement = getByText(/GitHub UserSearch/i);
  expect(headingElement).toBeInTheDocument();

  // Snapshot test: Capture and compare the rendered output
  expect(asFragment()).toMatchSnapshot();
});
