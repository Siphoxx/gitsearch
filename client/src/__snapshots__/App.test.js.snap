// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GitHub UserSearch heading 1`] = `
<DocumentFragment>
  <div>
    <h1>
      GitHub UserSearch
    </h1>
    <div
      class="user-search"
    >
      <input
        placeholder="Search for a GitHub user"
        type="text"
        value=""
      />
      <button>
        Search
      </button>
      <ul
        class="user-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
